File: php/lc00001-lc00100/lc00001/TwoSumTest.php
<?php

namespace leetcode\lc00001;

use PHPUnit\Framework\TestCase;

class TwoSumTest extends TestCase
{
    public function testTwoSum()
    {
        $cases = [
            ['nums' => [2, 7, 11, 15], 'target' => 9, 'expected' => [0, 1]],
            ['nums' => [2, 7, 11, 15], 'target' => 17, 'expected' => [0, 3]],
        ];

        $obj = new TwoSum();
        for ($i = 0; $i < count($cases); $i++) {
            $this->assertEquals($cases[$i]['expected'], $obj->twoSum($cases[$i]['nums'], $cases[$i]['target']));
        }
    }
}

File: golang/lc00001-lc00100/lc00001/two_sum_test.go
package lc00001

import (
	"testing"
)

type caseType struct {
	nums     []int
	target   int
	expected []int
}

var cases = []caseType{
	{[]int{2, 7, 11, 15}, 9, []int{0, 1}},
	{[]int{2, 7, 11, 15}, 17, []int{0, 3}},
}

func TestTwoSum(t *testing.T) {
	for _, c := range cases {
		result := twoSum(c.nums, c.target)
		if result[0] != c.expected[0] || result[1] != c.expected[1] {
			t.Fatalf("twoSum function failed, nums: %v, target: %d, execpted: %v, result: %v", c.nums, c.target, c.expected, result)
		}
	}
}

func TestTwoSumMap(t *testing.T) {
	for _, c := range cases {
		result := twoSumMap(c.nums, c.target)
		if result[0] != c.expected[0] || result[1] != c.expected[1] {
			t.Fatalf("twoSum function failed, nums: %v, target: %d, execpted: %v, result: %v", c.nums, c.target, c.expected, result)
		}
	}
}
